tests/ludwig/benchmarking/test_profiler.py

```diff
@@ -0,0 +1,61 @@
+import os
+import shutil
+import time
+
+import numpy as np
+import pandas as pd
+import torch
+
+from ludwig.api import LudwigModel
+from ludwig.benchmarking.profiler import LudwigProfiler
+from ludwig.constants import TRAINER
+
+
+def test_ludwig_profiler(tmpdir):
+    @LudwigProfiler(tag="test_function", output_dir=tmpdir, use_torch_profiler=False, logging_interval=0.1)
+    def func(duration):
+        time.sleep(duration)
+        x = torch.Tensor(2, 3)
+        y = torch.rand(2, 3)
+        torch.add(x, y)
+
+    train_df = pd.DataFrame(np.random.normal(0, 1, size=(100, 3)), columns=["input_1", "input_2", "output_1"])
+    eval_df = pd.DataFrame(np.random.normal(0, 1, size=(20, 3)), columns=["input_1", "input_2", "output_1"])
+
+    config = {
+        "input_features": [{"name": "input_1", "type": "number"}, {"name": "input_2", "type": "number"}],
+        "output_features": [{"name": "output_1", "type": "number"}],
+        "combiner": {"type": "concat", "output_size": 14},
+        TRAINER: {"epochs": 1},
+    }
+
+    model = LudwigModel(config=config, backend="local")
+
+    with LudwigProfiler(tag="train", output_dir=tmpdir, use_torch_profiler=False, logging_interval=0.1):
```

Comment:
nit: Use `profile_1`, `profile_2`, etc, to make the tag->os directory relationship more obvious.